File: BowlingGame/python/bowling_game_test.py
from .bowling_game import Game

def test_score_no_pins_hit():
    assert Game().score() == 0

def test_score_one_when_one_pin_hit():
    game = Game()
    game.roll(1)
    assert game.score() == 1

def test_score_handles_spares():
    game = Game()
    game.roll(3)
    game.roll(7)
    game.roll(4)
    assert game.score() == 18

def test_score_handles_strikes():
    game = Game()
    game.roll(10)
    game.roll(7)
    game.roll(2)
    assert game.score() == 28

def test_perfect_game():
    game = Game()
    for i in range(12):
        game.roll(10)
    assert game.score() == 300

def test_sample_chatgpt_game():
    game = Game()
    game.roll(1)
    game.roll(4)
    game.roll(4)
    game.roll(5)
    game.roll(6)
    game.roll(4)
    game.roll(5)
    game.roll(5)
    game.roll(10)
    game.roll(0)
    game.roll(1)
    game.roll(7)
    game.roll(3)
    game.roll(6)
    game.roll(4)
    game.roll(10)
    game.roll(2)
    game.roll(8)
    game.roll(6)
    assert game.score() == 133

File: BowlingGame/java/src/test/BowlingGameTest.java
package test;
import static org.junit.Assert.*;
import main.Game;

import org.junit.Before;
import org.junit.Test;

public class BowlingGameTest {

	private Game g;

	@Before
	public void setUp() {
		g = new Game();
	}
	
	@Test
	public void testGutterGame() {		
		rollMany(20, 0);
		assertEquals(0, g.score());
	}

	@Test
	public void testAllOnes() {		
		rollMany(20, 1);
		assertEquals(20, g.score());
	}
	
	@Test
	public void testOneSpare() {
		rollSpare();
		g.roll(3);
		rollMany(17, 0);
		assertEquals(16, g.score());
	}
	
	@Test
	public void testOneStrike() {
		rollStrike();
		g.roll(3);
		g.roll(4);
		rollMany(16,0);
		assertEquals(24, g.score());
	}

	@Test
	public void testPerfectGame() {
		rollMany(12, 10);
		assertEquals(300, g.score());
	}
	
	private void rollStrike() {
		g.roll(10);
	}
	
	private void rollSpare() {
		g.roll(5);
		g.roll(5);		
	}	
	
	private void rollMany(int rolls, int pins) {
		for (int i = 0; i < rolls; i++)
			g.roll(pins);
	}		
}
